File: src/main/java/dev/anvilcraft/addon/template/init/AddonItems.java
package dev.anvilcraft.addon.template.init;

import dev.anvilcraft.lib.v2.registrum.util.entry.ItemEntry;
import net.minecraft.world.item.Item;

import static dev.anvilcraft.addon.template.AnvilCraftAddonTemplate.REGISTRUM;

public class AddonItems {
    static {
        REGISTRUM.defaultCreativeTab(AddonItemGroups.ADDON_ITEMS.getKey());
    }

    public static final ItemEntry<Item> EXAMPLE_ITEM = REGISTRUM
        .item("example_item", Item::new)
        .register();

    public static void register() {
    }
}

File: src/main/kotlin/dev/anvilcraft/addon/template/init/AddonItems.kt
package dev.anvilcraft.addon.template.init

import dev.anvilcraft.addon.template.AnvilCraftAddonTemplate.Companion.REGISTRUM
import dev.anvilcraft.lib.v2.registrum.util.entry.ItemEntry
import net.minecraft.world.item.Item

class AddonItems {
    @Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
    companion object {
        init {
            REGISTRUM.defaultCreativeTab(AddonItemGroups.ADDON_ITEMS.key)
        }

        val EXAMPLE_ITEM: ItemEntry<Item> = REGISTRUM
            .item("example_item") { Item(it) }
            .register()

        fun register() {
        }
    }
}